lib/IRGen/GenCall.cpp

```diff
@@ -1684,19 +1687,21 @@ static void emitCoerceAndExpand(IRGenFunction &IGF,
   paramTI.deallocateStack(IGF, StackAddress(temporary), paramTy);
 }
 
-static void emitDirectExternalArgument(IRGenFunction &IGF, SILType argType,
-                                       llvm::Type *toTy, Explosion &in,
-                                       Explosion &out) {
+static void emitDirectExternalArgument(IRGenFunction &IGF,
+                                       const clang::CodeGen::ABIArgInfo &AI,
+                                       SILType argType, llvm::Type *argTy,
```

Comment:
The rename to 'argTy' seems strictly worse (for the local variable you should introduce instead of passing this separately).

Also, please just replace toTy in the parameter order; "source before target" is a nice thing to try to be consistent about.

Reply:
Replaced with a local `coercedTy` variable.  Sure, I can move the `ABIArgInfo` in the place of `toTy`, but that seems like additional information rather than source/target to me.

Reply:
It's the real "to type" information here: we don't need to produce a value of some abstract IR type, we need to produce a value matching the argument type scheme described by AI.